float32[] ranges
float32 dist_l
float32 dist_r
float32 min_depth
float32 width
int32 start_idx
int32 num_points


